lib/mix/tasks/popular_packages.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Mix.Tasks.PopularPackages do
+  use Mix.Task
+
+  @shortdoc "Rebuild popular_packages.txt"
+
+  @moduledoc """
+  Fetches a list of most downloaded packages from hex.pm and stores into the
+  top-level popular_packages.txt file.
+
+  ## Usage
+
+      mix popular_packages
+  """
+
+  @base_url "https://hex.pm/api/packages"
+  @file_path File.cwd!() <> "/popular_packages.txt"
+  @page_count 11
+
+  def run([]) do
+    Mix.shell().info("Fetching #{@page_count} pages of packages...")
+
+    names = build_popular_list()
+    count = length(names)
+
+    header = """
+    # The #{count} most popular packages on hex.pm, according to #{@base_url} on #{Date.utc_today() |> Date.to_string()}
+    # Generated using:
+    #     mix popular_packages
+    """
+
+    File.write!(@file_path, header <> Enum.join(names, "\n"))
+    Mix.shell().info("Wrote #{count} package names to #{@file_path}")
+  end
+
+  defp build_popular_list() do
+    1..@page_count
+    |> Enum.flat_map(&fetch_page/1)
+    |> Enum.map(& &1["name"])
```

Comment:
I love this task, but should we sort this list before writing it to a file so we don't modify the file just to re-order package names? I'm seeing many pointless "changes." For example:
```
-credo
 bunt
+credo
 recon
 websock
 telemetry_poller
@@ -46,16 +46,16 @@ finch
 phoenix_ecto
 expo
 jose
-nimble_parsec
 phoenix_live_view
+nimble_parsec
 httpoison
 verl
 tzdata
 tesla
 combine
 hex_core
-rebar3_hex
 phoenix_view
+rebar3_hex
 phoenix_live_dashboard
 dialyxir
 timex
@@ -69,9 +69,9 @@ earmark_parser
 makeup_elixir
 makeup
 joken
-ex_doc
-ex_aws
 gen_stage
+ex_aws
+ex_doc
```